crates/trie/src/walker.rs

```diff
@@ -235,7 +234,7 @@ impl<C: TrieCursor> TrieWalker<C> {
     }
 
     fn update_skip_node(&mut self) {
-        self.can_skip_current_node = if let Some(key) = self.key() {
+        self.can_skip_current_node = if let Some(key) = self.stack.last().map(|n| n.full_key()) {
```

Comment:
Is this not the same as `self.key()`?

Reply:
yes!
unfortunately line below does mutable borrow of self and if its done with key() method borrow checker things that the whole self is borrowed but when inlined this way it knows that we are working with different fields

a bit better look would be to move key method to self.stack, maybe?

Reply:
I see, this makes sense. Can also clean this up:
```diff
--- a/crates/trie/src/walker.rs
+++ b/crates/trie/src/walker.rs
@@ -234,10 +234,8 @@ impl<C: TrieCursor> TrieWalker<C> {
     }
 
     fn update_skip_node(&mut self) {
-        self.can_skip_current_node = if let Some(key) = self.stack.last().map(|n| n.full_key()) {
-            let contains_prefix = self.changes.contains(key);
-            let hash_flag = self.stack.last().unwrap().hash_flag();
-            !contains_prefix && hash_flag
+        self.can_skip_current_node = if let Some(node) = self.stack.last() {
+            node.hash_flag() && !self.changes.contains(node.full_key())
         } else {
             false
         };
```

Reply:
nice, committed your suggestion